app/components/primer/clipboard_copy.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use ClipboardCopy to populate copy values from the DOM
+  class ClipboardCopy < Primer::Component
+    status :alpha
+
+    # Optional target element that holds the data the user will copy.
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :target, lambda { |**system_arguments|
+      system_arguments[:tag] ||= :div
+      system_arguments[:id] = @id
+
+      Primer::BaseComponent.new(**system_arguments)
+    }
+
+    # @example Default
+    #   <%= render(Primer::ClipboardCopy.new(value: "Text to copy")) do %>
+    #     "Click to copy!"
+    #   <% end %>
+    #
+    # @example With a target
+    #   <%= render(Primer::ClipboardCopy.new(id: "foo")) do |component|
+    #     <%= component.target do %>
+    #       Text to copy
+    #     <% end %>
+    #
+    #     Click to copy!
+    #   <% end %>
+    #
+    # @example With a form target
+    #   <%= render(Primer::ClipboardCopy.new(id: "foo")) do |component|
+    #     <%= component.target(tag: :input, value: "Text to copy") %>
+    #
+    #     Click to copy!
+    #   <% end %>
+    #
+    # @example With a link target
+    #   <%= render(Primer::ClipboardCopy.new(id: "foo")) do |component|
+    #     <%= component.target(tag: :a, href: "/path/to/copy") %>
+    #
+    #     Click to copy!
+    #   <% end %>
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
```

Comment:
Might we add a `@param` for `value`, as we have an example above showing its use?